src/lib.rs

```diff
@@ -151,38 +150,52 @@ pub struct GptDisk {
 impl GptDisk {
     /// Add another partition to this disk.  This tries to find
     /// the optimum partition location with the lowest block device.
+    /// Returns the new partition id if there was sufficient room
+    /// to add the partition. Size is specified in bytes.
     pub fn add_partition(
         &mut self,
         name: &str,
         size: usize,
         part_type: partition_types::Type,
         flags: u64,
-    ) -> io::Result<()> {
+    ) -> io::Result<u32> {
+        let size_lba = match size.checked_div(self.config.lb_size.into()) {
```

Comment:
@cholcombe973 input parameter and intermediate computations should probably be `u64`.

Reply:
Yeah I was debating whether to use usize or u64.  Rust's std lib seems to prefer usize for io operations that's why i went with that.  I don't mind using u64